src/daemon/daemon.cpp

```diff
@@ -135,7 +135,7 @@ bool t_daemon::run(bool interactive)
 
     mp_internals->p2p.run(); // blocks until p2p goes down
 
-    if (interactive)
+    if (rpc_commands)
```

Comment:
It was undefined behavior during shutdown that we were likely getting lucky with sometimes. Or perhaps those weird reports of shutdown issues could have stemmed from this ...